internal/ingress/controller/nginx.go

```diff
@@ -745,3 +750,54 @@ func (n *NGINXController) setupSSLProxy() {
 		}
 	}()
 }
+
+// IsDynamicallyConfigurable decides if the new configuration can be dynamically configured without reloading
+func (n *NGINXController) IsDynamicallyConfigurable(pcfg *ingress.Configuration) bool {
+	runningBackends := make([]*ingress.Backend, 0, len(n.runningConfig.Backends))
+	newBackends := make([]*ingress.Backend, 0, len(pcfg.Backends))
+
+	for i := 0; i < len(n.runningConfig.Backends); i++ {
+		runningBackends = append(runningBackends, n.runningConfig.Backends[i].DeepCopy())
+	}
+
+	for i := 0; i < len(pcfg.Backends); i++ {
+		newBackends = append(newBackends, pcfg.Backends[i].DeepCopy())
+	}
+
+	n.runningConfig.Backends = []*ingress.Backend{}
```

Comment:
Please do not change the state of `n.runningConfig`. Please make a copy and use that to make the comparison.

Reply:
@aledbf notice that I'm changing only `n.runningConfig.Backends` and I have the copy of its original value at https://github.com/kubernetes/ingress-nginx/pull/2174/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R756 where I restore at https://github.com/kubernetes/ingress-nginx/pull/2174/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R772.

The idea is to make sure that the change between new and running config is only `Backends` change. Please let me know if you think there's a better way of doing this.

Reply:
You should not change `n.runningConfig`. I prefer a full copy that we can be disposed of later.

Reply:
Is there already a utility function to deep copy `runningConfig`?